ambry-cloud/src/test/java/com.github.ambry.cloud/CloudBlobStoreTest.java

```diff
@@ -255,6 +278,156 @@ public void testExceptionalDest() throws Exception {
     }
   }
 
+  /**
+   * Test PUT(with TTL) and TtlUpdate record replication.
+   * Replication may happen after PUT and after TtlUpdate, or after TtlUpdate only.
+   * PUT may already expired, expiration time < upload threshold or expiration time >= upload threshold.
+   * @throws Exception
+   */
+  @Test
+  public void testPutWithTtl() throws Exception {
+    // Set up remote host
+    MockClusterMap clusterMap = new MockClusterMap();
+    MockHost remoteHost = getLocalAndRemoteHosts(clusterMap).getSecond();
+    List<PartitionId> partitionIds = clusterMap.getWritablePartitionIds(null);
+    PartitionId partitionId = partitionIds.get(0);
+    StoreKeyFactory storeKeyFactory = new BlobIdFactory(clusterMap);
+    MockStoreKeyConverterFactory storeKeyConverterFactory = new MockStoreKeyConverterFactory(null, null);
+    storeKeyConverterFactory.setConversionMap(new HashMap<>());
+    storeKeyConverterFactory.setReturnInputIfAbsent(true);
+    MockStoreKeyConverterFactory.MockStoreKeyConverter storeKeyConverter =
+        storeKeyConverterFactory.getStoreKeyConverter();
+    Transformer transformer = new BlobIdTransformer(storeKeyFactory, storeKeyConverter);
+    Map<DataNodeId, MockHost> hosts = new HashMap<>();
+    hosts.put(remoteHost.dataNodeId, remoteHost);
+    MockConnectionPool connectionPool = new MockConnectionPool(hosts, clusterMap, 4);
+
+    // Generate BlobId for coming PUT.
+    short blobIdVersion = CommonTestUtils.getCurrentBlobIdVersion();
+    short accountId = Utils.getRandomShort(TestUtils.RANDOM);
+    short containerId = Utils.getRandomShort(TestUtils.RANDOM);
+    boolean toEncrypt = TestUtils.RANDOM.nextBoolean();
+    List<BlobId> blobIdList = new ArrayList<>();
+    for (int i = 0; i < 6; i++) {
+      blobIdList.add(
+          new BlobId(blobIdVersion, BlobId.BlobIdType.NATIVE, ClusterMapUtils.UNKNOWN_DATACENTER_ID, accountId,
+              containerId, partitionId, toEncrypt, BlobId.BlobDataType.DATACHUNK));
+    }
+
+    // Set up VCR
```

Comment:
Can any of this be merged into setupCloudStore or setBasicProperties, or is it too specific to the test?

Reply:
Fixed.